src/main/scala/rps/GameRepository.scala

```diff
@@ -1,44 +1,49 @@
 package rps
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext
 import java.util.UUID
 import java.sql.Timestamp
 import java.time.Instant
 
 import slick.driver.H2Driver.backend.DatabaseDef
 import slick.driver.H2Driver.api._
-
-import db.Tables.{Plays, PlayRow}
+import db.Tables.{PlayRow, Plays}
 import model._
+import zio.{IO, Task}
 
 trait GameRepository {
-  def save(play: Play): Future[Either[Throwable, UUID]]
-  def read(): Future[Either[Throwable, Option[Play]]]
+  def save(play: Play): IO[RPSError, UUID]
+
+  def read(): IO[RPSError, Option[Play]]
 }
 
 class GameRepositoryImpl(
   db: DatabaseDef
 )(
   implicit ec: ExecutionContext
-) extends GameRepository with SlickRepository {
+) extends GameRepository {
 
-  override def save(play: Play): Future[Either[Throwable, UUID]] = {
+  override def save(play: Play): IO[RPSError, UUID] = {
     val playRow = convertPlay(play)
     val newPlay = Plays += playRow
-    
-    futureToEither(db.run(newPlay).map(_ => playRow.id))
+
+    IO.fromFuture { _ => db.run(newPlay) }
+      .as(playRow.id)
+      .mapError(e => RPSError.DBError(e.getMessage))
   }
 
-  override def read(): Future[Either[Throwable, Option[Play]]] = { 
+  override def read(): IO[RPSError, Option[Play]] = {
     val selectPlay = Plays.sortBy(_.createdAt.desc).take(1).result.headOption
-    futureToEither(db.run(selectPlay).map(_.flatMap(convertPlayRow)))
-  } 
+    val maybePlayRowIO: Task[Option[PlayRow]] = IO.fromFuture { _ => db.run(selectPlay) }
+    maybePlayRowIO.map(_.flatMap(convertPlayRow))
+  }.mapError(e => RPSError.DBError(e.getMessage))
```

Comment:
```suggestion
	IO.fromFuture { _ => db.run(selectPlay) }
	  .map(_.flatMap(convertPlayRow))
	  .mapError(e => RPSError.DBError(e.getMessage)) 
  }
```

Reply:
variable names were meant to help a newcomer to understand this passage deeply, step-by-step

Reply:
I am the first fun of oneliners, but... I think when you are learning, even if you can read them, you don't stop to think about what's happening inside the chain

Reply:
ok, however I think they're a bit confusing if that's the purpose. I don't understand what `maybePlayRowIO` is by just reading the name and it feels a bit arbitrary. Anyway, I don't mind, we can leave it as is

Reply:
I'm with @gabro. I'm not a newcomer and I struggled to understand the naming